tests/e2e/lib/virt_helpers.go

```diff
@@ -513,6 +581,155 @@ func (v *VirtOperator) ensureHcoRemoved(timeout time.Duration) error {
 	return err
 }
 
+func (v *VirtOperator) getVmStatus(namespace, name string) (string, error) {
+	vm, err := v.Dynamic.Resource(virtualMachineGvr).Namespace(namespace).Get(context.Background(), name, v1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+
+	status, ok, err := unstructured.NestedString(vm.UnstructuredContent(), "status", "printableStatus")
+	if err != nil {
+		return "", err
+	}
+	if !ok {
+		return "", fmt.Errorf("status field not populated yet on VM %s/%s", namespace, name)
+	}
+	log.Printf("VM %s/%s status is: %s", namespace, name, status)
+
+	return status, nil
+}
+
+func (v *VirtOperator) checkVmExists(namespace, name string) bool {
+	_, err := v.getVmStatus(namespace, name)
+	if err == nil {
+		return true
+	}
+	return false
+}
+
+func (v *VirtOperator) checkVmStatus(namespace, name, expectedStatus string) bool {
+	status, _ := v.getVmStatus(namespace, name)
+	return status == expectedStatus
+}
+
+func (v *VirtOperator) createVm(namespace, name, source string) error {
+	unstructuredVm := unstructured.Unstructured{
+		Object: map[string]interface{}{
```

Comment:
Is there a type for this in virt repos?

If yes, I would import here (if this does not increase size of OADP binary) or copy it here

Reply:
Yes, recently I tried importing the kubevirt modules but gave up trying to resolve all the conflicts in a reasonable amount of time. I would like to fix the module conflicts and replace all the unstructured stuff, but for now this lets me get the actual test working quicker.